File: VB/MyXtraGrid/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace MyXtraGrid
    Partial Public Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Private customers As BindingList(Of Customer)
        Public Sub New()
            InitializeComponent()
            FillGridDataSource()
            InitGridView()
        End Sub
        Private Sub InitGridView()
            myGridView1.ColumnPanelRowHeight = verticalColumnHeaderPanel
            AddHandler myGridView1.CustomDrawColumnHeader, AddressOf myGridView1_CustomDrawColumnHeader
        End Sub
        Private verticalColumnHeaderPanel As Integer = 85
        Private Sub myGridView1_CustomDrawColumnHeader(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.ColumnHeaderCustomDrawEventArgs)
            If e.Column IsNot Nothing AndAlso (Not e.Column.Visible) Then
                Return
            End If
            DrawVertical(e)
        End Sub

        Private Shared Sub DrawVertical(ByVal e As DevExpress.XtraGrid.Views.Grid.ColumnHeaderCustomDrawEventArgs)
            Dim r As Rectangle
            r = e.Info.CaptionRect
            e.Info.Caption = String.Empty
            e.Painter.DrawObject(e.Info)
            If e.Column IsNot Nothing Then
                Dim state = e.Graphics.Save()
                Dim sf = New StringFormat()
                sf.Trimming = StringTrimming.EllipsisCharacter
                sf.FormatFlags = sf.FormatFlags Or StringFormatFlags.NoWrap
                sf.FormatFlags = sf.FormatFlags Or StringFormatFlags.DirectionVertical
                e.Graphics.DrawString(e.Column.GetTextCaption(), e.Appearance.Font, e.Appearance.GetForeBrush(e.Cache), r, sf)
                e.Graphics.Restore(state)
            End If
            e.Handled = True
        End Sub
        Private Sub FillGridDataSource()
            customers = New BindingList(Of Customer)()
            For i = 0 To 49
                Dim customer = New Customer()
                customer.Name = "Mike"
                If i Mod 2 = 0 Then
                    customer.Name = "John"
                End If
                customer.Age = 70 - i
                customer.Weight = 50 + i
                customer.Height = 150 + i
                customers.Add(customer)
            Next i
            myGridControl1.DataSource = customers
        End Sub
    End Class
End Namespace

File: CS/MyXtraGrid/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;

namespace MyXtraGrid {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        private BindingList<Customer> customers;
        public Form1() {
            InitializeComponent();
            FillGridDataSource();
            InitGridView();
        }
        private void InitGridView() {
            myGridView1.ColumnPanelRowHeight = verticalColumnHeaderPanel;
            myGridView1.CustomDrawColumnHeader += myGridView1_CustomDrawColumnHeader;
        }
        private int verticalColumnHeaderPanel = 85;
        private void myGridView1_CustomDrawColumnHeader(object sender, DevExpress.XtraGrid.Views.Grid.ColumnHeaderCustomDrawEventArgs e) {            
            if (e.Column != null && !e.Column.Visible) {
                return;
            }
            DrawVertical(e);
        }

        private static void DrawVertical(DevExpress.XtraGrid.Views.Grid.ColumnHeaderCustomDrawEventArgs e) {
            Rectangle r;
            r = e.Info.CaptionRect;
            e.Info.Caption = string.Empty;
            e.Painter.DrawObject(e.Info);
            if(e.Column != null) {
                var state = e.Graphics.Save();
                var sf = new StringFormat();
                sf.Trimming = StringTrimming.EllipsisCharacter;
                sf.FormatFlags |= StringFormatFlags.NoWrap;
                sf.FormatFlags |= StringFormatFlags.DirectionVertical;
                e.Graphics.DrawString(e.Column.GetTextCaption(), e.Appearance.Font, e.Appearance.GetForeBrush(e.Cache), r, sf);
                e.Graphics.Restore(state);
            }
            e.Handled = true;
        }
        private void FillGridDataSource() {
            customers = new BindingList<Customer>();
            for (var i = 0; i < 50; i++) {
                var customer = new Customer();
                customer.Name = "Mike";
                if (i % 2 == 0) {
                    customer.Name = "John";
                }
                customer.Age = 70 - i;
                customer.Weight = 50 + i;
                customer.Height = 150 + i;
                customers.Add(customer);
            }
            myGridControl1.DataSource = customers;
        }
    }
}
